Assets/dbConn.cs

```diff
@@ -4,32 +4,84 @@
 using Mono.Data.Sqlite;
 using System.Data;
 using System;
+using UnityEngine.UI;
 
 public class dbConn : MonoBehaviour
 {
-    void Start()
+    private string conn;
+
+    private List<HighScore> highScores = new List<HighScore>();
+
+    public GameObject scorePrefab;
+
+    public Transform scoreParent;
+
+    public int topRanks;
+
+    public void Start()
+    {
+        highScores.Clear();
+
+        conn = "URI=file:" + Application.dataPath + "/db/db.db";
+
+        ShowScores();
+    }
+
+    //Update ranking after click Ranking button - needed?
+    /*
+    public void EnterRanking()
+    {
+        highScores.Clear();
+
+        Debug.Log(highScores.Count);
+
+        conn = "URI=file:" + Application.dataPath + "/db/db.db";
+
+        ShowScores();
+    }
+    */
+    public void ReadScores() 
     {
-        string conn = "URI=file:" + Application.dataPath + "/db/db.db"; //Path to database.
-        IDbConnection dbconn;
-        dbconn = (IDbConnection)new SqliteConnection(conn);
-        dbconn.Open(); //Open connection to the database.
-        IDbCommand dbcmd = dbconn.CreateCommand();
-        string sqlQuery = "SELECT * FROM SCORES";
-        dbcmd.CommandText = sqlQuery;
-        IDataReader reader = dbcmd.ExecuteReader();
-        while (reader.Read())
+        
+        using (IDbConnection dbconn = new SqliteConnection(conn))
         {
-            int value = reader.GetInt32(0);
-            int score = reader.GetInt32(1);
-            Debug.Log("id_score= " + value + "  score =" + score);
+            dbconn.Open();
+
+            using (IDbCommand dbcmd = dbconn.CreateCommand())
+            {
+                string sqlQuery = "SELECT * FROM SCORES";
+                dbcmd.CommandText = sqlQuery;
+
+                using (IDataReader reader = dbcmd.ExecuteReader())
+                {
+                    while (reader.Read())
+                    {
+                        highScores.Add(new HighScore(reader.GetInt32(0), reader.GetInt32(1), reader.GetString(2), reader.GetString(3)));
+                    }
+                    dbconn.Close();
+                    reader.Close();
+                }
+            }
         }
-        reader.Close();
-        reader = null;
-        dbcmd.Dispose();
-        dbcmd = null;
-        dbconn.Close();
-        dbconn = null;
+        highScores.Sort();
     }
 
+    private void ShowScores()
+    {
+        ReadScores();
+        for (int i = 0; i < topRanks; i++)
+        {
+            if (i <= highScores.Count - 1)
+            {
+
+                GameObject tmpObject = Instantiate(scorePrefab);

```

Comment:
Let's call it something like `scoreView`. I think in the future `tmpObject` name may be confusing.

Reply:
Change to scoreView